src/equations/compressible_euler_multicomponent_1d.jl

```diff
@@ -202,16 +202,15 @@ function initial_condition_weak_blast_wave(x, t,
     r = abs(x_norm)
     cos_phi = x_norm > 0 ? 1 : -1
 
-    prim_rho = SVector{ncomponents(equations), real(equations)}(r > 0.5f0 ?
-                                                                2^(i - 1) * (1 - 2) /
-                                                                (1 -
-                                                                 2^ncomponents(equations)) *
-                                                                one(RealT) :
```

Comment:
This version from me is wrong as `2^(i - 1) * (1 - 2) /(1 - 2^ncomponents(equations))` brings some `Float64` computation.